bundles/org.openhab.binding.shelly/README.md

```diff
@@ -0,0 +1,514 @@
+If you are using password protected  devices you need to configure user id and password. 
+A special thing is created if the access to the device's APU cannot be initialized.
+You need to edit the Thing configuration and setup user id and password (same as in the Shelly App) to activate this Thing.
+The Thing will be re-initialized and the Thing type adjusted and this was successful.
+
+## Binding Configuration
+
+The binding has some global configuration options.
+
+| Parameter      |Description                                                       |Mandatory|Default                                         |
+|----------------|------------------------------------------------------------------|---------|------------------------------------------------|
+| defaultUserId  |Default user id for HTTP authentication when not set in the Thing |    no   |admin                                           |
+| defaultPassword|Default password for HTTP authentication when not set in the Thing|    no   |admin                                           |
+
+## Thing Configuration
+
+|Parameter         |Description                                                   |Mandatory|Default                                           |
+|------------------|--------------------------------------------------------------|---------|--------------------------------------------------|
+|deviceIp          |IP address of the Shelly device                               |    yes  |none                                              |
+|userId            |The user id used for HTTP authentication                      |    no   |none                                              |
+|password          |Password for HTTP authentication*                             |    no   |none                                              |
+|lowBattery        |Threshold for battery level. Set alert when level is below.   |    no   |20 (=20%), only for battery powered devices       |
+|updateInterval    |Interval for the background status check in seconds.          |    no   |1h for battery powered devices, 60s for all others|
+|eventsButton      |true: register event "trigger when a button is pushed"        |    no   |false                                             |
+|eventsPush        |true: register event "trigger on short and long push"         |    no   |false                                             |
+|eventsSwitch      |true: register event "trigger of switching the relay output"  |    no   |true                                              |
+|eventsSensorReport|true: register event "posted updated sensor data"             |    no   |true for sensor devices                           |
+|eventsCoIoT       |true: Listen for CoIoT/COAP events                            |    no   |true for battery devices, false for others        |
+
+Independent from the updateInterval setting the binding will perform a settings refresh from the device once per minute.
```

Comment:
This I do not understand. Does it mean that those are not configuration values of the handler, but something that is stored in the device? If so, what should happen if the handler is configured textually? Usually, values that are in the device are modelled as channels. Which parameters are these here?

Reply:
you mean this "Independent from the updateInterval setting the binding will perform a settings refresh from the device once per minute."?

You could set the status update interval (polling when not using Coap). However, to get settings, which were changed in the Apps the binding refreshes the settings once per minute even if status interval is < 60s.

Reply:
or the event options (eventsSensorReport...):
This allows to disable setting the event urls (e.g. if you have another controller already set those urls)

Reply:
I mean that configuration parameters are there to configure the HANDLER, not the remote device. See also https://github.com/eclipse/smarthome/issues/3484.

Reply:
no, those options configure the handler
if there is more than „one“ controller the user could decide, which one receives the event callback
another reasons: Shelly still has a problem when enabling all events. This could lead into a device reboot. With 1.5.6 it‘s better, but still this could happen (that was also the reason to implement the uptime monitoring being able to quantify this situation). Sometimes a http could also lead into a timeout, I increased it to 6s, which made it better, but still..
All of that is reported to Shelly, but so far...

Reply:
Hm, I still do not understand how your infos are related to "perform a settings refresh". If those settings (=parameters) are options of the handler, why does it require any refresh from the device?